presto-native-execution/presto_cpp/external/dsdgen/dsdgen-c/address.cpp

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright owned by the Transaction Processing Performance Council.
+ *
+ * A copy of the license is included under external/dsdgen/LICENSE
+ * in this repository.
+ *
+ * You may not use this file except in compliance with the License.
+ *
+ * THE TPC SOFTWARE IS AVAILABLE WITHOUT CHARGE FROM TPC.
+ */
+
+#include "address.h"
+#include <stdio.h>
+#include "columns.h"
+#include "config.h"
+#include "dist.h"
+#include "genrand.h"
+#include "permute.h"
+#include "porting.h"
+#include "r_params.h"
+#include "scaling.h"
+#include "tables.h"
+#include "tdefs.h"
+
+static int s_nCountyCount = 0;
+static int s_nCityCount = 0;
+
+void resetCountCount(void) {
+  s_nCountyCount = 0;
+  s_nCityCount = 0;
+
+  return;
+}
+
+/*
+ * Routine:
+ * Purpose:
+ * Algorithm:
+ * Data Structures:
+ *
+ * Params:
+ * Returns:
+ * Called By:
+ * Calls:
+ * Assumptions:
+ * Side Effects:
+ * TODO: None
+ */
+int mk_address(ds_addr_t* pAddr, int nColumn, DSDGenContext& dsdGenContext) {
+  int i, nRegion;
```

Comment:
Lets initialize variables before their first usage, unless they are set in the very next call e.g. like it is done for nMaxCities and nMaxCounties.